nav2_behavior_tree/plugins/condition/goal_updated_condition.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2020 Aitor Miguel Blanco
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__GOAL_UPDATED_CONDITION_HPP_
```

Comment:
Can you add to the readme for BT navigator about how to implement preemption using this workflow? And then show examples of a tree using this to do a preemption, I think its important to make it explicitly clear to BT designers their needs for getting this behavior. The planning and recovery cases make good case studies.

Reply:
Yes! I knew I was forgetting something.